### PR TITLE
SLING-10084 Prevent the creation of features with multiple identical …

### PR DESCRIPTION
**Pull request by bosschaert**
…Configuration PIDs

### CODE REVIEW

File: src/test/java/org/apache/sling/feature/io/json/ConfigurationJSONWriterTest.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.Test;
+
+import java.io.CharArrayWriter;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static org.junit.Assert.fail;
+
+public class ConfigurationJSONWriterTest {
+    @Test
```
Comment (rombert) [inline src/test/java/org/apache/sling/feature/io/json/ConfigurationJSONWriterTest.java:30]: Maybe use `@Test(expected = IOException.class)` ? Makes the code clearer IMO.

**Comment by bosschaert** [comment]
I understand that at least there is also an issue about duplicate keys within a single PID. Will update the PR for that shortly.

**Comment by cziegeler** [comment]
Patch looks good to me, not sure if this is part of this issue or another one, but we should handle this when reading from a feature file as well and fail reading on duplicate keys

**Comment by cziegeler** [comment]
@bosschaert After reading the spec again, I can't find that pids are used case-insensitive. Properties are
And the config admin implementation compares pids case-sensitive, so I think we should not add this logic

**Comment by bosschaert** [comment]
You are right @cziegeler, I thought that PIDs were also case-insensitive, but I checked the spec too and could not find this mentioned. As you say it's only the property keys that are case-insensitive.
I'll close this PR so.

**Pull request closed without merging**